File: CS/CS_Group2/Member.cpp
#include <iostream>
#include <fstream>
#include <vector> //for precised arrays.
#include <iomanip>
#include <string>
#include <cstdlib>
#include <limits>
#include <algorithm> // to find the value in range
using namespace std;

class Member {
protected:
    string name;
    int memberId;
    vector<int> borrowedBooks; // Store book IDs borrowed by this member
public:
    Member(string n, int id) : name(n), memberId(id) {}

    virtual void displayMember() {
        cout << setw(5) << memberId << setw(20) << name << endl;
    }

    void addBorrowedBook(int bookId) {
        borrowedBooks.push_back(bookId);
    }

    void returnBorrowedBook(int bookId) {
        auto it = find(borrowedBooks.begin(), borrowedBooks.end(), bookId);
        if (it != borrowedBooks.end()) {
            borrowedBooks.erase(it);
        }
    }

    void displayBorrowedBooks() {
        cout << "Books borrowed by " << name << " (ID: " << memberId << "):\n";
        if (borrowedBooks.empty()) {
            cout << "No books borrowed.\n";
        } else {
            for (int bookId : borrowedBooks) {
                cout << "Book ID: " << bookId << endl;
            }
        }
    }

    int getId() { return memberId; }
};

File: Acity_JAVA_Project/Warren Ojbonnaya ESONU.10012300011/Member.java
import java.io.*;
import java.util.*;

public class Member {
    protected String name;
    protected int memberId;
    protected List<Integer> borrowedBooks = new ArrayList<>();

    public Member(String name, int memberId) {
        this.name = name;
        this.memberId = memberId;
    }

    // Getters
    public int getId() {
        return memberId;
    }

    // Method to display member details
    public void displayMember() {
        System.out.printf("%-5d %-20s%n", memberId, name);
    }

    // Add borrowed book to the list
    public void addBorrowedBook(int bookId) {
        borrowedBooks.add(bookId);
    }

    // Return borrowed book from the list
    public void returnBorrowedBook(int bookId) {
        borrowedBooks.remove(Integer.valueOf(bookId));
    }

    // Display list of borrowed books
    public void displayBorrowedBooks() {
        try (FileWriter fw = new FileWriter("displayborrowedbook.txt", true); BufferedWriter bw = new BufferedWriter(fw)) {
            bw.write("Books borrowed by " + name + " (ID: " + memberId + "):\n");

            if (borrowedBooks.isEmpty()) {
                bw.write("No books borrowed.\n");
            } else {
                for (int bookId : borrowedBooks) {
                    bw.write("Book ID: " + bookId + "\n");
                }
            }
            bw.newLine();
        } catch (IOException e) {
            System.out.println("Error writing to file.");
        }
    }
}